cargo-dylint/tests/ci.rs

```diff
@@ -133,6 +133,165 @@ fn cargo_dylint_and_dylint_readmes_are_equal() {
     compare_lines(&cargo_dylint_readme, &dylint_readme);
 }
 
+#[test]
+fn examples_readme_contents() {
+    let examples_dir = find_examples_dir();
+    let categories = vec![
+        "general",
+        "supplementary",
+        "restriction",
+        "experimental",
+        "testing",
+    ];
+    
+    // Read the existing README
+    let readme_path = examples_dir.join("README.md");
+    let readme_content = read_to_string(&readme_path).unwrap();
+
+    // Generate just the lint description tables
+    let tables = generate_lint_tables(&examples_dir, &categories);
+    
+    // Extract the current tables section from README using markers
+    let current_tables = extract_between_markers(&readme_content, 
+        "<!-- lint descriptions start -->",
+        "<!-- lint descriptions end -->"
+    ).unwrap_or_else(|| panic!("Lint description markers not found in README.md"));
+
+    // Extract and compare just the meaningful content (links and descriptions)
+    let expected_content = extract_meaningful_content(&tables);
+    let actual_content = extract_meaningful_content(&current_tables);
+
+    assert_eq!(
+        expected_content,
+        actual_content,
```

Comment:
```suggestion
        expected_tables,
        actual_tables,
```
I would prefer this more stringent check. Please also remove the `extract_meaningful_content` function.

I understand why you wrote the `extract_meaningful_content` function: to avoid having to worry about spacing.

But generating a table with correct spacing will make it easier to replace an existing, incorrect table.